lib/rules/test-case-property-ordering.js

```diff
@@ -31,29 +31,59 @@ module.exports = {
             (tests || []).forEach(test => {
-              const keys = (test.properties || []).map(utils.getKeyName);
+              const properties = test.properties || [];
+              const keyNames = properties.map(utils.getKeyName);
 
-              for (let i = 0, lastChecked; i < keys.length; i++) {
-                const current = order.indexOf(keys[i]);
+              for (let i = 0, lastChecked; i < keyNames.length; i++) {
+                const current = order.indexOf(keyNames[i]);
 
                 // current < lastChecked to catch unordered;
                 // and lastChecked === -1 to catch extra properties before.
                 if (current > -1 && (current < lastChecked || lastChecked === -1)) {
-                  let orderMsg = order.filter(item => keys.indexOf(item) > -1);
+                  let orderMsg = order.filter(item => keyNames.indexOf(item) > -1);
                   orderMsg = orderMsg.concat(
-                    lastChecked === -1 ? keys.filter(item => order.indexOf(item) === -1) : []
+                    lastChecked === -1 ? keyNames.filter(item => order.indexOf(item) === -1) : []
                   );
 
                   context.report({
-                    node: test.properties[i],
+                    node: properties[i],
                     message,
                     data: { order: orderMsg.join(', ') },
+                    fix (fixer) {
+                      const source = sourceCode.getText();
+                      const last = properties[properties.length - 1];
+                      // end loc to replace(including whitespaces).
+                      const finalEnd = sourceCode.getTokenAfter(last, token => token.value === '}' && token.type === 'Punctuator').range[0];
+                      // reorder the properties and put in result array.
+                      const result = [];
+                      for (let j = 0; j < keyNames.length; j++) {
+                        const insertedIndex = orderMsg.indexOf(keyNames[j]);
+                        const propertyCode = sourceCode.getText(properties[j]);
+                        const propertyStart = properties[j].range[1];
+                        const propertyEnd = j < properties.length - 1 ? properties[j + 1].range[0] : finalEnd;
+                        let trailing = source.slice(propertyStart, propertyEnd);
+
+                        // for last property, should check & add trailling commas.
+                        if (j === properties.length - 1 && sourceCode.getTokenAfter(last).value !== ',') {
+                          trailing = ', ' + trailing;
+                        }
+                        result[insertedIndex] = propertyCode + trailing;
+                      }
+
+                      const start = properties[0].range[0];
+
+                      return fixer.replaceTextRange(
+                        [start, finalEnd],
+                        result.join('')
+                      );
+                    },
```

Comment:
Would it be possible to use the [new feature in ESLint 4.1.0](http://eslint.org/blog/2017/06/eslint-v4.1.0-released#applying-multiple-autofixes-simultaneously) where you can return an array of autofixes?

That way, this autofix would be very simple, because it wouldn't be necessary to deal with whitespace or trailing commas:

```js
fix(fixer) {
  return orderMsg.map((key, index) => {
    const propertyToInsert = test.properties[keyNames.indexOf(key)];
    return fixer.replaceText(test.properties[index], sourceCode.getText(propertyToInsert));
  });
}
```